### PR TITLE
ctre: added releases up to 3.2

### PR DESCRIPTION
**Pull request by axalon900**
Specify library name and version:  **ctre/3.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by axalon900** [comment]
Right, something failed and I have no idea what if anything needs fixing. Every version packaged fine locally with hooks passing.

File: recipes/ctre/all/conandata.yml [hunk 1]
```diff
@@ -2,3 +2,33 @@ sources:
...
+    sha256: b1f24383a4c12c68a9da6e61283e02c1e6ff27875af33a8a95ccca72256b4dc9
+  "3.0.1":
+    url: https://github.com/hanickadot/compile-time-regular-expressions/archive/v3.0.1.tar.gz
+    sha256: 0838ce0af9e9e89ffc10854dc98c88b2dd40d1866ceaf0dc6c1663d6c39882cc
+  "3.1":
+    url: https://github.com/hanickadot/compile-time-regular-expressions/archive/v3.1.tar.gz
+    sha256: c209ab0533b56a4fbbc7959e10c91c89fe2c9cb1725b64ed52be272a534a1d83
+  "3.2":
+    url: https://github.com/hanickadot/compile-time-regular-expressions/archive/v3.2.tar.gz
+    sha256: d4e734e3dd501f741424aceb5198f5ebbc35c624335c99f7e202a9ecb31cd3e6
```
Comment (Croydon) [inline recipes/ctre/all/conandata.yml:34]: This is quite excessive. Please only add versions you actually need

**Comment by Croydon** [comment]
> Right, something failed and I have no idea what if anything needs fixing. 

This is an unexpected error. When there are regular errors the bot would posts a comment with links to logs

Closing and re-opening the pull request would trigger the bot again, but so does a new commit

Please add only versions you actually need. Assuming that you don't need all versions, a new commit removing some of them will trigger the CI again and hopefully it works this time 😄

**Review by Croydon — APPROVED** [review]

File: recipes/ctre/all/conandata.yml (hunk 1 above)
Comment (axalon900) [reply recipes/ctre/all/conandata.yml:22]: I've cut out all but the latest of each major/minor.

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**